lib/core/auth/auth_provider.js

```diff
@@ -10,136 +8,20 @@ const MongoError = require('../error').MongoError;
 class AuthProvider {
   constructor(bson) {
     this.bson = bson;
```

Comment:
We have a bit of an inconsistency here that might be fine to leave in since its been the existing behavior but only AWS auth uses this bson property. Meanwhile PLAIN inits its own BSON so it ignores user overrides. Since the work to take out bson will have to be done in the port to master, if done here it could make the porting work easier.

Reply:
Yea that does sound correct so no need to make changes here.